lib/routes/geocaching/blogs.ts

```diff
@@ -21,35 +35,82 @@ export const route: Route = {
         },
     ],
     name: 'Official Blogs',
-    maintainers: ['HankChow'],
+    maintainers: ['HankChow', 'Konano'],
     handler,
     url: 'geocaching.com/blog/',
 };
 
+const languageToCategory = { de: 140, fr: 138, es: 702, nl: 737, cs: 1404 };
+const languageToLabel = { de: 'Deutsch', fr: 'Français', es: 'Español', nl: 'Nederlands', cs: 'Čeština' };
+
 async function handler(ctx) {
     const baseUrl = 'https://www.geocaching.com';
-    const { data: response } = await got(`${baseUrl}/blog/wp-json/wp/v2/posts`, {
-        searchParams: {
-            per_page: ctx.req.query('limit') ?? 100,
-            _embed: 1,
-        },
-    });
+    const language = ctx.req.param('language') ?? 'en';
+    const searchParams: {
+        per_page: number;
+        _embed: number;
+        _fields: string;
+        categories_exclude?: string;
+        categories?: number;
+    } = {
+        per_page: 20,
+        _embed: 1,
+        _fields: ['id', 'title', 'link', 'guid', 'excerpt', 'date_gmt', 'modified_gmt', '_embedded', '_links'].join(','),
+    };
+
+    if (language === 'en') {
+        searchParams.categories_exclude = Object.values(languageToCategory).join(',');
+    } else if (language in languageToCategory) {
+        searchParams.categories = languageToCategory[language];
+    } else if (language === 'all') {
+        // do nothing
+    } else {
+        throw new Error(`Unsupported language: ${language}`);
+    }
 
-    const items = response.map((item) => ({
-        title: item.title.rendered,
-        link: item.link,
-        guid: item.guid.rendered,
-        description: item.content.rendered,
-        pubDate: parseDate(item.date_gmt),
-        author: item._embedded.author[0].name,
-        category: item._embedded['wp:term'][0].map((category) => category.name),
-    }));
+    // console.log(searchParams);
+
+    const { data: response } = await got(`${baseUrl}/blog/wp-json/wp/v2/posts`, { searchParams });
+    const items = response.map((item) => {
+        const media = item._embedded['wp:featuredmedia'][0];
+        const mediaDetails = media?.media_details;
+        return {
+            title: item.title.rendered.trim(),
+            link: item.link,
+            guid: item.guid.rendered,
+            description: item.excerpt.rendered,
```

Comment:
Why remove the ability to retrieve full article?

Reply:
The [specification](https://www.rssboard.org/rss-specification#hrelementsOfLtitemgt) defines the description of `item` as "The item synopsis."

Another reason is that the full article is too large.

Reply:
fixed in https://github.com/DIYgod/RSSHub/commit/df7a5467f454f242c80288a285279f68b4d979e0